lib/iris/cube.py

```diff
@@ -1632,7 +1638,38 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
-        self._dtype = dtype
+        if dtype != self.dtype:
+            if dtype is not None:
+                if not self.has_lazy_data():
+                    emsg = 'Cube does not have lazy data, cannot set dtype.'
+                    raise ValueError(emsg)
+                dtype = np.dtype(dtype)
+                if dtype.kind != 'i':
+                    emsg = ('Can only cast lazy data to integral dtype, '
+                            'got {!r}.')
```

Comment:
Just a note: if you're visiting this bit, I've found that this test should also be allowing 'u' type data, i.e. unsigned ints.  E.G. `if dtype.kind not in ('i', 'u'):` ?
If not it can wait -- I think it could do with a more general code search anyway, as I think we may have missed this possibility elsewhere in the code too.